inquire/src/prompts/path_select/mod.rs

```diff
@@ -0,0 +1,503 @@
+mod action;
+use action::*;
+mod prompt;
+use prompt::*;
+mod config;
+use config::*;
+
+use crate::{
+    config::get_configuration,
+    error::InquireResult,
+    formatter::MultiOptionFormatter,
+    list_option::ListOption,
+    prompts::prompt::Prompt,
+    terminal::get_default_terminal,
+    type_aliases::Filter,
+    ui::{Backend, MultiSelectBackend, RenderConfig},
+    InquireError,
+};
+use std::{
+    convert::TryFrom,
+    ffi::OsStr,
+    fmt, fs,
+    ops::Deref,
+    path::{Path, PathBuf},
+};
+
+/// Different path selection modes specify what the user can choose
+#[derive(Clone, Eq, PartialEq)]
+pub enum PathSelectionMode<'a> {
+    /// The user may pick a file with the given (optional) extension
+    File(Option<&'a str>),
+    /// The user may pick a directory
+    Directory,
+    /// The user may pick multiple paths
+    Multiple(Vec<PathSelectionMode<'a>>),
+}
+impl<'a> Default for PathSelectionMode<'a> {
+    fn default() -> Self {
+        Self::Directory
+    }
+}
+
+/// Path with cached information
+#[derive(Clone, Debug, Hash)]
+pub struct PathEntry {
```

Comment:
I think the amount of PathEntry impls warrant a new module just for PathEntry, living alongside this current file.